src/sinks/websocket/sink.rs

```diff
@@ -261,6 +261,18 @@ impl WebSocketSink {
 
         let bytes_sent = register!(BytesSent::from(Protocol("websocket".into())));
         let events_sent = register!(EventsSent::from(Output(None)));
+        let encode_as_binary = match self.encoder.serializer() {
+            codecs::encoding::Serializer::RawMessage(_)
+            | codecs::encoding::Serializer::Avro(_)
+            | codecs::encoding::Serializer:: Native(_) => true,
+            codecs::encoding::Serializer::Csv(_)
+            | codecs::encoding::Serializer::Logfmt(_)
+            | codecs::encoding::Serializer::Gelf(_)
+            | codecs::encoding::Serializer::Json(_)
+            | codecs::encoding::Serializer::Text(_)
+            | codecs::encoding::Serializer:: NativeJson(_) => false,
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏With listing all of these out, it should clean it up quite nicely to add a `use` statement for `codecs::encoding::Serializer` up at the top of the file.

Reply:
Getting real nitpicky now, but my preference would be to extract as a function and then add the `use` statement local to the function. Definitely not blocking though 😄

Reply:
updated